uplink/ecclient/client.go

```diff
@@ -473,30 +474,81 @@ func (lr *lazyPieceRanger) Size() int64 {
 // Range implements Ranger.Range to be lazily connected
 func (lr *lazyPieceRanger) Range(ctx context.Context, offset, length int64) (_ io.ReadCloser, err error) {
 	defer mon.Task()(&ctx)(&err)
-	ps, err := lr.dialPiecestore(ctx, &pb.Node{
+
+	reader := &lazyPieceReader{}
+	reader.lazyPieceRanger = lr
+	reader.ctx = ctx
+	reader.offset = offset
+	reader.length = length
```

Comment:
I don't know what our style guide is, but I personally prefer:

```
reader := &lazyPieceReader{
    ranger: lr,
    ctx: ctx,
    etc....
}
```

rather than setting each variable individually after the fact.